libraries/wallet/wallet.cpp

```diff
@@ -2013,9 +2054,14 @@ class wallet_api_impl
          vbos = _remote_db->get_vesting_balances( *acct_id );
          if( vbos.size() == 0 )
          {
-            witness_object wit = get_witness( account_name );
-            FC_ASSERT( wit.pay_vb );
-            vbid = wit.pay_vb;
+            if (is_witness(account_name)) 
```

Comment:
I don't think the `withdraw_GPOS_vesting_balance` users will be witnesses. They could be but they dont need to. Assert msg only should change here IMHO.

Reply:
If I don't keep this condition then it will fail with "No account or witness named" which doesn't sound relevant for this operation.

Reply:
The failure you see is in `get_witness`. We should not call it at all and go only with the account. Witnesses can withdraw from their vesting balance with `withdraw_vesting`.

Reply:
Make sense, I will remove checking witness vesting from `withdraw_GPOS_vesting_balance`. I've also added the same condition in `withdraw_vesting`. I will keep it as it is there.

Reply:
`withdraw_vesting` is used to withdraw normal vesting only and not the gpos vesting. Witnesses having gpos vesting will have to call `withdraw_GPOS_vesting_balance` so I think I will have to keep this part unchanged.

Reply:
I think that will never happen. In order for the witnesses to have gpos vesting balance he will have to create and deposit by operation. Payments from the network to the witnesses are created automatically by the blockchain and they are of the default type(normal) with a cdd policy. Witness withdraw this by using `withdraw_vesting`.

A witness can have gpos vesting balance but this will be a separated balance from the network payment. In order to withdraw from these the witness will use `withdraw_GPOS_vesting_balance`.

At the end `withdraw_GPOS_vesting_balance` is totally independent from if the user is a witness so the check can be removed.